### PR TITLE
Fix local environment variable setup

### PR DESCRIPTION
**Pull request by tdeebswihart**
**What changed?**

References to the local IP were fixed in our test environment setup code. I also tidied up env var initialization while I was there and logged the errors we receive from os.Setenv when it fails.

**Why?**

As part of #4766 we accidentally replaced uses of the hard-coded IPv4 localhost value with the environment variable we use to inject the local IP. This broke our ability to run tests against locally-running dependencies

**How did you test it?**

I ran our integration and functional tests against locally-running cassandra, mysql, and postgresql servers.

**Potential risks**

Presumably nothing; environment.SetupEnv() is only called by test code.

**Is hotfix candidate?**
No

### CODE REVIEW

**Review by stephanos — APPROVED** [review]
Love it!

File: environment/env.go [hunk 1]
```diff
@@ -70,75 +70,61 @@ const (
...
-
-	if os.Getenv(ESSeeds) == "" {
-		err := os.Setenv(ESSeeds, LocalhostIP)
-		if err != nil {
-			panic(fmt.Sprintf("error setting env %v", ESSeeds))
-		}
-	}
+type varSpec struct {
+	Name    string
+	Default func() string
```
Comment (bergundy) [inline environment/env.go:75]: nit: make these private

**Review by bergundy — APPROVED** [review]
Approved with a nit. Thanks!

File: environment/env.go (hunk 1 above)
Comment (tdeebswihart) [reply environment/env.go]: Done! `Default` is now `getDefault` as `default` is a reserved word

**Pull request merged**